### PR TITLE
T14847

### PR DESCRIPTION
**Pull request by joaquimrocha**
https://phabricator.endlessm.com/T14847

### CODE REVIEW

File: src/eos-google-chrome-app [hunk 1]
```diff
@@ -151,8 +133,42 @@ class GoogleChromeFlatpakLauncher:
             if not runtime_path or not os.path.exists(runtime_path):
                 exit_with_error("Could not find Chrome's external runtime directory")
 
-        # Both application and runtime directories found if reached.
-        app_info = {'app_lib_path': app_lib_path, 'runtime_path': runtime_path}
+            runtime_path = os.path.join(runtime_path, 'files')
+            if os.path.exists(runtime_path):
```
Comment (mariospr) [inline src/eos-google-chrome-app]: This looks like it should be `if not ...`

File: src/eos-google-chrome-app [hunk 2]
```diff
@@ -151,8 +133,42 @@ class GoogleChromeFlatpakLauncher:
...
+            exit_with_error("Could not find Chrome's application directory")
+
+        app_lib_path = os.path.join(app_path, 'files/lib/x86_64-linux-gnu')
+        if not os.path.exists(app_lib_path):
+            exit_with_error("Could not find bundled libraries for Chrome")
+
+        if app_path and not app_lib_path:
+            exit_with_error("Chrome's installation directory found, but not bundled libraries")
+
+
```
Comment (mariospr) [inline src/eos-google-chrome-app]: nit: This extra blank line is not needed

File: src/eos-google-chrome-app [hunk 3]
```diff
@@ -151,8 +133,42 @@ class GoogleChromeFlatpakLauncher:
...
+        chrome_path = os.path.join(app_path, 'files', 'extra')
+        # We check for legacy runtimes to maintain backward-compatibility for users without
+        # support for native external apps (Flatpak <0.6.13)
+        if not os.path.exists(chrome_path):
+            exit_with_error("Could not find 'extra data' directory in Chrome's Flatpak; falling "
```
Comment (mariospr) [inline src/eos-google-chrome-app]: You should not exit here, or you'll never return the legacy path. `logging.warning()` would probably be better

File: src/eos-google-chrome-app [hunk 4]
```diff
@@ -108,26 +110,7 @@ class GoogleChromeFlatpakLauncher:
...
-            exit_with_error("Could not find Chrome's application directory")
-
-        app_lib_path = os.path.join(app_path, 'files/lib/x86_64-linux-gnu')
-        if not os.path.exists(app_lib_path):
-            exit_with_error("Could not find bundled libraries for Chrome")
-
-        if app_path and not app_lib_path:
-            exit_with_error("Chrome's installation directory found, but not bundled libraries")
-
+    def _get_legacy_chrome_path(self, app_path, app_info):
```
Comment (mariospr) [inline src/eos-google-chrome-app]: maybe a better name would be `_get_legacy_chrome_runtime_path()`, but no strong feelings

File: src/eos-google-chrome-app (hunk 1 above)
Comment (joaquimrocha) [inline src/eos-google-chrome-app]: Good catch!

File: src/eos-google-chrome-app (hunk 3 above)
Comment (joaquimrocha) [inline src/eos-google-chrome-app]: Yikes.

**Comment by mariospr** [comment]
Thanks for addressing those comments, looking good now

**Pull request merged**